### PR TITLE
Patch ctypes.util.find_library to support Emscripten platform

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Several packages use `ctypes.util.find_library` to search shared libraries. But it does not support Emscripten platform. I opened a upstream PR about this (https://github.com/python/cpython/pull/99950), but it isn't accepted as CPython WASM build has no support for ctypes now.

This PR adds a patch that adds Emscripten platform support for ctypes.util.find_library: searching libraries from paths that ar in LD_LIBRARY_PATH env variable.

Note that [Emscripten uses LD_LIBRARY_PATH for `dlopen` internally](https://github.com/emscripten-core/emscripten/blob/efc18ae6e268373ff984d1622979422f99f3db0a/src/library_dylink.js#L965-L968), though I think it is not documented :/

This is also required for #3292 

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by rth** [comment]
Looks very nice, thank ! I'll let Hood approve.

File: src/js/dynload.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ function createDynlibFS(
   readFileFunc?: ReadFileType,
 ): LoadDynlibFS {
   const dirname = lib.substring(0, lib.lastIndexOf("/"));
-  const defaultSearchDirs = [API.dsodir, API.sitepackages, dirname];
+  const defaultSearchDirs = API.defaultLdLibraryPath + [dirname];
```
Comment (hoodmane) [inline src/js/dynload.ts:35]: You can't add lists like this in Typescript can you? You are probably thinking of Python.

**Review by hoodmane — APPROVED** [review]
Generally looks good to me though I am confused about list addition in typescript (see comment).

File: src/js/dynload.ts (hunk 1 above)
Comment (ryanking13) [reply src/js/dynload.ts]: Good point, Thanks! It seems like this was the reason why I was getting mysterious errors 😂

**Pull request merged**